src/main/webapp/app/shared/util/firebase/firebase-crud-store.ts

```diff
@@ -70,14 +71,23 @@ export class FirebaseCrudStore<T> {
     }
   }
 
+  async updateUntemplated(path: string, value: any) {
+    const convertValue = convertNestedObject(value);
+    try {
+      return await update(ref(this.db, path), convertValue);
+    } catch (e) {
+      notifyError(e, 'Error updating to Firebase');
+    }
+  }
+
   delete(path: string) {
     return remove(ref(this.db, path));
   }
 
   async deleteFromArray(path: string, indices: number[]) {
     return await runTransaction(ref(this.db, path), (currentData: any[]) => {
       const newData = [];
-      for (let i = 0; i < currentData.length; i++) {
+      for (let i = 0; currentData !== null && i < currentData.length; i++) {
```

Comment:
In what scenario this will ben null?

Reply:
Honestly, I do not remember. I think I must've encountered it for me to have put that there. However, I can think of a scenario off the top of my head:

Since it uses `runTransaction` any changes to the currentdata will result in the transaction being rerun with updated currentdata. Thus, if someone deletes all comments while you are deleting a comment, currentData can become null when the function is rerun. 

Wanted to get this comment in before the end of the day since you won't be in tomorrow, but I will try to reproduce to remind myself why I originally put that there.